examples/composition/apps/src/main.rs

```diff
@@ -0,0 +1,349 @@
+
+    // Build the proof requirements with the specified selector
+    let mut requirements = Requirements::new(image_id, Predicate::digest_match(journal.digest()));
+    if disable_aggregation {
+        requirements = requirements.with_unaggregated_proof();
+    }
+
+    // Build the proof request offer
+    let offer = Offer::default()
+        // The market uses a reverse Dutch auction mechanism. Set min and max prices per million cycles.
+        .with_min_price_per_mcycle(parse_ether("0.001")?, mcycles_count)
+        // NOTE: If your offer is not being accepted, try increasing the max price.
+        .with_max_price_per_mcycle(parse_ether("0.002")?, mcycles_count)
+        // Timeouts for the request and lock.
+        .with_timeout(1000)
+        .with_lock_timeout(1000);
+
+    // Build and submit the request
+    let request = ProofRequest::builder()
+        .with_image_url(image_url)
+        .with_input(input_url)
+        .with_requirements(requirements)
+        .with_offer(offer)
+        .build()?;
+    let (request_id, expires_at) = client.submit_request(&request).await?;
+    tracing::info!("Request {} submitted", request_id);
+
+    // Wait for the request to be fulfilled (check periodically)
+    tracing::info!("Waiting for request {} to be fulfilled", request_id);
+    let (_, seal) = client
+        .wait_for_request_fulfillment(
+            request_id,
+            Duration::from_secs(5), // periodic check every 5 seconds
+            expires_at,
+        )
+        .await?;
+    tracing::info!("Request {} fulfilled", request_id);
+
+    Ok((journal, seal))
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use alloy::{
+        network::EthereumWallet,
+        node_bindings::{Anvil, AnvilInstance},
+        providers::{ProviderBuilder, WalletProvider},
+    };
+    use boundless_market::contracts::{
+        hit_points::default_allowance,
+        test_utils::{create_test_ctx, TestCtx},
+    };
+    use boundless_market::storage::MockStorageProvider;
+    use broker::test_utils::BrokerBuilder;
+    use guest_assessor::ASSESSOR_GUEST_ID;
+    use guest_set_builder::SET_BUILDER_ID;
+    use test_log::test;
+    use tokio::task::JoinSet;
+
+    alloy::sol!(
+        #![sol(rpc)]
+        Counter,
+        "../contracts/out/Counter.sol/Counter.json"
+    );
+
+    async fn deploy_counter<P: Provider + 'static + Clone + WalletProvider>(
+        anvil: &AnvilInstance,
+        test_ctx: &TestCtx<P>,
+    ) -> Result<Address> {
+        let deployer_signer: PrivateKeySigner = anvil.keys()[0].clone().into();
+        let deployer_provider = ProviderBuilder::new()
+            .wallet(EthereumWallet::from(deployer_signer))
+            .connect(&anvil.endpoint())
+            .await
+            .unwrap();
+        let counter = Counter::deploy(&deployer_provider, test_ctx.set_verifier_address).await?;
+        Ok(*counter.address())
+    }
+
+    #[test(tokio::test)]
+    #[ignore = "does not work with RISC0_DEV_MODE=1"]
```

Comment:
This means that this test does not currently run in the CI. However, the test works with the local prover (on x86) and Bonsai, so we could switch the CI to use it.